solarforecastarbiter/reference_forecasts/persistence.py

```diff
@@ -362,3 +365,244 @@ def _check_intervals_times(interval_label, data_start, data_end,
+        have the same units as the observation data (e.g. MW).
+
+    Raises
+    ------
+    ValueError
+        If there is insufficient data for matching by time of day or the
+        **axis** parameter is invalid.
+
+    Notes
+    -----
+    Assumes that there is at least 20 days of *observation* data available
+    based on [1]_, [2]_, [3]_.
+
+    References
+    ----------
+    .. [1] Allessandrini et al. (2015) "An analog ensemble for short-term
+       probabilistic solar power forecast", Appl. Energy 157, pp. 95-110.
+       doi: 10.1016/j.apenergy.2015.08.011
+    .. [2] Yang (2019) "A universal benchmarking method for probabilistic
+       solar irradiance forecasting", Solar Energy 184, pp. 410-416.
+       doi: 10.1016/j.solener.2019.04.018
+    .. [3] Doubleday, Van Scyoc Herndandez and Hodge (2020) "Benchmark
+       probabilistic solar forecasts: characteristics and recommendations",
+       Solar Energy 206, pp. 52-67. doi: 10.1016/j.solener.2020.05.051
+
+    See also
+    --------
+    :py:func:`solarforecastarbiter.reference_forecasts.persistence.persistence_probabilistic`
+
+    """
+    # ensure that we're using times rounded to multiple of interval_length
+    _check_intervals_times(observation.interval_label, data_start, data_end,
+                           forecast_start, forecast_end,
+                           observation.interval_length)
+
+    closed = datamodel.CLOSED_MAPPING[interval_label]
+    fx_index = pd.date_range(start=forecast_start, end=forecast_end,
+                             freq=interval_length, closed=closed)
+
+    # observation data resampled to match forecast sampling
+    obs = load_data(observation, data_start, data_end)
+    obs = obs.resample(interval_length, closed=closed).mean()
+
+    # time of day: minutes past midnight (e.g. 0=12:00am, 75=1:15am)
+    obs_timeofday = (obs.index.hour * 60 + obs.index.minute).astype(int)
+    fx_timeofday = (fx_index.hour * 60 + fx_index.minute).astype(int)
+
+    # confirm sufficient data for matching by time of day
+    if data_end - data_start < pd.Timedelta('20D'):
+        raise ValueError("Insufficient data to match by time of day")
+
+    if axis == "x":
+        forecasts = []
+        for constant_value in constant_values:
+            fx = pd.Series(np.nan, index=fx_index)
+            for tod in fx_timeofday:
+                data = obs[obs_timeofday == tod]
+                cdf = ECDF(data)
+                fx[fx_timeofday == tod] = cdf(constant_value) * 100.0
+            forecasts.append(fx)
+    elif axis == "y":
+        forecasts = []
+        for constant_value in constant_values:
+            fx = pd.Series(np.nan, index=fx_index)
+            for tod in fx_timeofday:
+                data = obs[obs_timeofday == tod]
+                fx[fx_timeofday == tod] = np.percentile(data, constant_value)
+            forecasts.append(fx)
+    else:
+        raise ValueError(f"Invalid axis parameter: {axis}")
```

Comment:
missing from coverage. Add a simple test that this is raised

Reply:
Added a test.